qemu/tests/hpt_miscellaneous.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import re
+
+from virttest import error_context
+from virttest import utils_misc
+
+
+def set_hpt(vm, test, hpt_size):
+    """Run this step to set HPT in guest"""
+    session = vm.wait_for_login(timeout=600)
+    set_cmd = "echo %s > /sys/kernel/debug/powerpc/hpt_order"
+    try:
+        st, o = session.cmd_status_output(set_cmd % (hpt_size), timeout=60)
+        n_output = str(o)
+    except Exception as detail:
+        if st == 1:
+            # valid parameters,hpt_size > = 18 and hpt_size < = 46
+            if hpt_size >= 18 and hpt_size <= 46:
+                if re.search(r"Operation not permitted", n_output):
+                    pass
+                elif re.rearch(r"No space left on device", n_output):
+                    pass
+                else:
+                    test.fail("Valid param but: '%s'" % n_output)
+            # invalid parameter 0,it is handled in different way
+            # by kernel so dealing with it particularly
+            elif hpt_size == 0:
+                if not re.search(r"Input/output error", n_output):
+                    test.fail("Invalid param zero : '%s'" % n_output)
+            # invalid parameters,hpt_size < 18 and hpt_size > 46
+            else:
+                if not re.search(r"Invalid argument", n_output):
+                    test.fail("Invalid param: '%s'" % n_output)
+
+
+def get_hpt(test, params, session, hpt_size, get_cmd):
+    """Run this step to check HPT in guest"""
+    if params.get("mistype") != "negative":
+        get_hpt_value = session.cmd_output(get_cmd, timeout=60)
+        if int(get_hpt_value) != hpt_size:
+            test.fail("HPT expected results aren't consistent")
+
+
+def mem_sub_hpt(test, vm):
+    """Run this step to hot plug 1G memory to guest"""
+    error_context.context("hotplug 1G memory before resizing HPT")
+    mem_id = utils_misc.generate_random_string(8)
+    dimm_id = utils_misc.generate_random_string(8)
+    vm.monitor.cmd_qmp("object-add", {
+                           "qom-type": "memory-backend-ram",
+                           "id": "%s" % mem_id,
+                           'props': {'policy': 'default', 'size': 1073741824}})
+    vm.monitor.cmd_qmp("device_add", {
+                           "driver": "pc-dimm",
+                           "id": "%s" % dimm_id,
+                           "memdev": "%s" % mem_id})
```

Comment:
Please do these steps by using the functions provided by the vt framework, here is an example:
https://github.com/avocado-framework/avocado-vt/blob/db3d67afcb11728d194838a3d8f01c0f137ed864/virttest/utils_test/qemu/__init__.py#L609-L627

Reply:
Updated this part already and thanks for the reference.